src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateSet.cs

```diff
@@ -26,36 +16,10 @@ private sealed class StateSet
             public readonly DiagnosticAnalyzer Analyzer;
             public readonly bool IsHostAnalyzer;
 
-            private readonly ConcurrentSet<DocumentId> _activeDocuments;
-            private readonly ConcurrentDictionary<ProjectId, ProjectState> _projectStates;
```

Comment:
ProjectState ends up not being needed at all, so all this tracking of them goes away.  similarly, _activeDocuments ends up not actually ever being read due to other changes in this PR.  So that data goes away.  So 'StateSet' is just hte Analyzer and the boolean property above.  Note: i will rename this type ina  followup, but didn't here to keep the diff smaller.